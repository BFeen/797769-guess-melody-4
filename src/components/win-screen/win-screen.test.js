import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import WinScreen from "./win-screen.jsx";
import history from "../../history.js";


describe(`WinScreen snapshot testing`, () => {
  it(`WinScreen rendering correctly with 3 questions`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <WinScreen
            questionsCount={3}
            mistakesCount={3}
            onReplayButtonClick={() => {}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`WinScreen rendering correctly with 1 mistake`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <WinScreen
            questionsCount={3}
            mistakesCount={1}
            onReplayButtonClick={() => {}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe(`With 2 questions`, () => {
    it(`With 0 mistake`, () => {
      const tree = renderer.create(
          <Router history={history}>
            <WinScreen
              questionsCount={2}
              mistakesCount={0}
              onReplayButtonClick={() => {}}
            />
          </Router>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`With 1 mistake`, () => {
      const tree = renderer.create(
          <Router history={history}>
            <WinScreen
              questionsCount={2}
              mistakesCount={1}
              onReplayButtonClick={() => {}}
            />
          </Router>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
