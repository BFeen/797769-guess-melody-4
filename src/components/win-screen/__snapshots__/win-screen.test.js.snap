// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinScreen snapshot testing WinScreen rendering correctly with 1 mistake 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Правильных ответов: 
    2
    . Количество ошибок: 
    1
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`WinScreen snapshot testing WinScreen rendering correctly with 3 questions 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Правильных ответов: 
    0
    . Количество ошибок: 
    3
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`WinScreen snapshot testing With 2 questions With 0 mistake 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Правильных ответов: 
    2
    . Количество ошибок: 
    0
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`WinScreen snapshot testing With 2 questions With 1 mistake 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Правильных ответов: 
    1
    . Количество ошибок: 
    1
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;
